move goals to separate service

mkhomishchak opened the pull request:
also I added temporary API for user's balance, once we will start using kafka for sync calls it can be removed

DendeberiaOleksandr commented on src/main/java/com/khomishchak/ws/controllers/BalanceController.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    @GetMapping("/transactions-history/period")
+    public Double getDepositAmountForPeriod(@RequestParam Long userId, @RequestParam String ticker,
```
It could be more flexible if you make it possible to not bound itself by providing date range as required parameter. The reasons to bound API like this one are:
1) You have a big amount of data 
2) You have sensitive data
I think it's not a case of any. So you can move it to simple `/transactions/history` endpoint. Set date range as optional parameters. It will make your API more flexible.

DendeberiaOleksandr commented on src/main/java/com/khomishchak/ws/services/exchangers/ExchangerService.java:
```diff
@@ -20,6 +21,8 @@ public interface ExchangerService {
 
     List<Balance> getAllMainBalances(long userId);
 
+    double getDepositValueForPeriod(long userId, String ticker, LocalDateTime startingDate, LocalDateTime endingDate);
```
Adjust to above. You can use Criteria API(Specification) or make your own for filtering.

DendeberiaOleksandr commented on src/main/java/com/khomishchak/ws/services/exchangers/ExchangerServiceImpl.java:
```diff
@@ -66,6 +70,25 @@ public List<Balance> getAllMainBalances(long userId) {
+    private double getDepositValueForPeriodForSingleIntegratedBalance(ExchangerDepositWithdrawalTransactions transactions,
+                                                                      String ticker, LocalDateTime startingDate,
+                                                                      LocalDateTime endingDate) {
+        return transactions.getTransactions().stream()
```
Adjust to nullable date parameters

DendeberiaOleksandr commented on src/test/java/com/khomishchak/ws/repositories/SelfGoalRepositoryTest.java:
```diff
@@ -1,75 +1,75 @@
-    }
-}
\ No newline at end of file
+//package com.khomishchak.ws.repositories;
```
:)

DendeberiaOleksandr commented on src/test/java/com/khomishchak/ws/services/goal/impl/DepositSelfGoalValidatorTest.java:
```diff
@@ -1,101 +1,101 @@
-    }
-}
\ No newline at end of file
+//package com.khomishchak.ws.services.goal.impl;
```
:))

DendeberiaOleksandr commented on src/test/java/com/khomishchak/ws/services/scheduled/GoalsScheduledServiceTest.java:
```diff
@@ -1,108 +1,108 @@
-    }
-}
\ No newline at end of file
+//package com.khomishchak.ws.services.scheduled;
```
:))

mkhomishchak committed:
made transactions history sum amount API more flexible

made params for getting total amount of ticker for transactions history api more flexible by removing required passing of the filtration params
used Search criteria for more flexible filtration of all type of transactions(criteria can be expanded with more arguments if needed)
added handler for different type of transation results (COMPLETED, DENIED and so on) for WBT integration
